src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -880,40 +898,52 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			connection.options['azureAccountToken'] = tokenPromise.token;
 			return true;
 		}
+		let account = this.findAzureAccountFromProfile(accounts, connection);
+		if (account) {
+			this._logService.debug(`Getting security token for Azure account ${account.key.accountId}`);
+			if (account.isStale) {
+				this._logService.debug(`Account is stale - refreshing`);
+				try {
+					account = await this._accountManagementService.refreshAccount(account);
+				} catch (err) {
+					this._logService.info(`Exception refreshing stale account : ${toErrorMessage(err, true)}`);
+					// refreshAccount throws an error if the user cancels the dialog
+					return false;
+				}
+			}
+			const tenantId = connection.azureTenantId;
+			const token = await this._accountManagementService.getAccountSecurityToken(account, tenantId, azureResource);
+			this._logService.debug(`Got token for tenant ${token}`);
```

Comment:
token is an object, did you verify if this actually prints out the information correctly?

But also we should generally never be printing out the actual token to the logs (that's very sensitive information). I'd suggest making the message as `Got token for tenant ${tenantId}`